lib/rest-adapter.js

```diff
@@ -365,49 +366,7 @@ RestAdapter.errorHandler = function(options) {
         // the handler are reported
         err = handlerError;
       }
-      if (typeof err === 'string') {
-        err = new Error(err);
-        err.status = err.statusCode = 500;
-      }
-
-      if (res.statusCode === undefined || res.statusCode < 400) {
-        res.statusCode = err.statusCode || err.status || 500;
-      }
-
-      if (Array.isArray(err)) {
-        var details = err.map(function(it) {
-          var data = generateResponseError(it);
-          delete data.statusCode;
-          return data;
-        });
-
-        var msg = 'Failed with multiple errors, see `details` for more information.';
-        err = new Error(msg);
-        err.details = details;
-      }
-
-      res.send({ error: generateResponseError(err) });
-
-      function generateResponseError(error) {
-        debug('Error in %s %s: %s', req.method, req.url, error.stack);
-
-        var data = {
-          name: error.name,
-          status: res.statusCode,
-          message: error.message || 'An unknown error occurred',
-        };
-
-        for (var prop in error) {
-          data[prop] = error[prop];
-        }
-
-        data.stack = error.stack;
-        if (process.env.NODE_ENV === 'production' || options.disableStackTrace) {
-          delete data.stack;
-        }
-
-        return data;
-      }
+      return new strongErrorHandler(options)(err, req, res, next);
```

Comment:
@davidcheung I would like you to take an extra step and warn users upgrading from 2.x to 3.0 when they forget to change the configuration of the error handler and keep using `option.disableStackTrace` (which no longer works).

My idea is modify `RestAdapter.errorHandler` and add a bit of code at the beginning that will detect unused option(s):

```js
RestAdapter.errorHandler = function(options) {
  options = options || {};
  if ('disableStackTrace' in options) {
    console.warn(
      'strong-remoting no longer supports "errorHandler.disableStackTrace" option. ' +
      'Use the new option "errorHandler.debug" instead.');
  }
  // etc.
};
```